br/pkg/streamhelper/advancer.go

```diff
@@ -544,13 +553,43 @@ func (c *CheckpointAdvancer) subscribeTick(ctx context.Context) error {
 	return c.subscriber.PendingErrors()
 }
 
+func (c *CheckpointAdvancer) isCheckpointLagged(ctx context.Context) (bool, error) {
+	if c.cfg.CheckPointLagLimit <= 0 {
+		return false, nil
+	}
+
+	now, err := c.env.FetchCurrentTS(ctx)
+	if err != nil {
+		return true, err
```

Comment:
We may encounter some temporary errors when fetching TSO from a remote node. Maybe always return `false` when `err` isn't `nil`?

Reply:
That may lead to some unexpected problem when the errors are not just "temporary", but I that make sense.